Gordon360/Services/AdministratorService.cs

```diff
@@ -12,72 +13,58 @@ namespace Gordon360.Services
     public class AdministratorService : IAdministratorService
     {
         private CCTContext _context;
+        private IAccountService _accountService;
 
-        public AdministratorService(CCTContext context)
+        public AdministratorService(CCTContext context, IAccountService accountService)
         {
             _context = context;
+            _accountService = accountService;
         }
-
         /// <summary>
-        /// Fetches the admin resource whose id is specified as an argument.
+        /// Fetches all the administrators from the database
         /// </summary>
-        /// <param name="id">The admin ID.l</param>
-        /// <returns>The Specified administrator. If none was found, a null value is returned.</returns>
-        public ADMIN Get(int id)
+        /// <returns>Returns a list of administrators. If no administrators were found, an empty list is returned.</returns>
+        public IEnumerable<AdminViewModel> GetAll()
         {
-            var query = _context.ADMIN.Find(id);
-            if (query == null)
-            {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The Administrator was not found." };
-            }
-            return query;
+            return _context.ADMIN.Select<ADMIN,AdminViewModel>(a => a);
         }
 
         /// <summary>
-        /// Fetches the admin resource whose username matches the specified argument
-        /// </summary>
-        /// <param name="gordon_id">The administrator's gordon id</param>
-        /// <returns>The Specified administrator. If none was found, a null value is returned.</returns>
-        public ADMIN Get(string gordon_id)
-        {
-            var query = _context.ADMIN.FirstOrDefault(x => x.ID_NUM.ToString() == gordon_id);
-            if (query == null)
-            {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The Administrator was not found." };
-            }
-            return query;
-        }
-        /// <summary>
-        /// Fetches all the administrators from the database
+        /// Fetches a specific admin from the database
         /// </summary>
         /// <returns>Returns a list of administrators. If no administrators were found, an empty list is returned.</returns>
-        public IEnumerable<ADMIN> GetAll()
+        public AdminViewModel? GetByUsername(string username)
         {
-            return _context.ADMIN;
+            var admin = _context.ADMIN.FirstOrDefault(a => a.USER_NAME == username);
+            if (admin != null)
+                return admin;
+            return null;
```

Comment:
The `if` check is kinda redundant - 'if A is not null return, otherwise return null' simplifies to just 'return A', and since the nullability is encoded in the Type System, we aren't losing any information by doing that.

Reply:
Ah, I just changed this from a cast and didn't actually look at it afterwards (it was only casting if admin was not null.

Reply:
Nope, nevermind. I was apparently right to leave the if according to VS:
![image](https://user-images.githubusercontent.com/43764277/196228514-70d29798-59ec-41ec-95cf-bd49bc72607c.png)

Reply:
Ah, that's because the implicit conversion from `ADMIN` to `AdminViewModel` doesn't handle nulls. 

Two ways to alleviate that: overload the implicit conversion operator to accept `ADMIN?` and return `AdminViewModel?`, or change this method to be `_context.ADMIN.FirstOrDefault(...).Select<ADMIN, AdminViewModel>();`

Reply:
I like the first one. I just didn't know how to do it

Reply:
Done.